infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -865,6 +955,64 @@ private Map<UUID, StreamAddressSpace> getStreamsAddressesMap(
         return requestedAddressSpaces;
     }
 
+    /**
+     * Get a list of streams backed by which the tables have been dropped, including their CP streams
+     *
+     * @return a list of all inactive streams
+     */
+    private List<UUID> getDroppedStreams() {
+
+        if (!isRegistryTableOpened())
+        {
+            return Collections.emptyList();
+        }
+
+        CorfuRuntime corfuRuntime = getNewCorfuRuntime();
+
+        PersistentCorfuTable<CorfuStoreMetadata.TableName,
+                CorfuRecord<CorfuStoreMetadata.TableDescriptors, CorfuStoreMetadata.TableMetadata>>
+                registry = corfuRuntime.getTableRegistry().getRegistryTable();
+
+        List<UUID> realStreams = registry.entryStream()
+            .filter(entry -> entry.getValue().getMetadata().getIsDropped())
+            .map(entry -> CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(
+                entry.getKey().getNamespace(), entry.getKey().getTableName())))
+            .collect(Collectors.toList());
+        List<UUID> checkpointStreams = realStreams.stream()
+            .map(CorfuRuntime::getCheckpointStreamIdFromId)
+            .collect(Collectors.toList());
+
+        List<UUID> droppedStreams = new ArrayList<>();
+        droppedStreams.addAll(realStreams);
+        droppedStreams.addAll(checkpointStreams);
+
+        corfuRuntime.shutdown();
+        return droppedStreams;
+    }
+
+    private CorfuRuntime getNewCorfuRuntime() {
+        final CorfuRuntime.CorfuRuntimeParameters params =
+                serverContext.getManagementRuntimeParameters();
+        params.setSystemDownHandlerTriggerLimit(SYSTEM_DOWN_HANDLER_TRIGGER_LIMIT);
+        final CorfuRuntime runtime = CorfuRuntime.fromParameters(params);
+        final Layout managementLayout = serverContext.copyManagementLayout();
+        // Runtime can be set up either using the layout or the bootstrapEndpoint address.
+        if (managementLayout != null) {
+            managementLayout.getLayoutServers().forEach(runtime::addLayoutServer);
+        }
+        runtime.connect();
+        log.info("getCorfuRuntime: Corfu Runtime connected successfully");
+        params.setSystemDownHandler(runtimeSystemDownHandler);
+        return runtime;
+    }
+
+    private boolean isRegistryTableOpened() {
```

Comment:
I think the nomenclature can be improved further for clarity. Consider using initialized instead of Opened?

Reply:
Done.